src/services/game-logic/pieces/Pawn.ts

```diff
@@ -1,36 +1,32 @@
 import { PieceNames } from '../../../enums';
 import { Piece } from '../pieces/Piece';
-import { ColorsEnum } from '../ColorsEnum';
-import { Move } from '../../../models/Move';
+import { Colors } from '../../../enums/Colors';
+import { Square } from '../../../models/Square';
 
 /**
  * This class implements Pawn behavoiur in chess
- * @class
  */
 class Pawn extends Piece {
     name = PieceNames.PAWN;
 
     /**
      * Returs possible moves of Pawn.
-     * @returns {Array}
      */
-    getPossibleMoves(): Move[] {
-        const move = new Move();
-        move.from.row = this.position.row;
-        move.from.column = this.position.column;
+    getPossibleMoves(): Square[] {
+        const move = new Square(-1, -1);
```

Comment:
Nie musisz tworzyć obiektu z modelu, typescriptowi wystarczy że struktura obiektu jest taka sama więc możesz zwracać od razu: `{column: this.position.column, row: this.position +1}` ale to tylko tak informacyjnie jak chcesz wykorzystywać konstruktor to spoko.

Reply:
Zostawię w ten sposób, ale dziękuję za tipa